### DISCUSSION TITLE
Can I request the VS dev cmd environment when not using cl.exe as the compiler?

### DISCUSSION THREAD

**Discussion by tristanlabelle** [microsoft/vscode-cmake-tools#3486] [Q&A] [answered]
Hello!

I'm building a project with a `CMakePresets.json` file which specifies the `clang` compiler. This seems to turn off the CMake Tools' setting up of a VS dev command prompt environment variables before calling `cmake.exe`, since it then fails to find `ninja` whereas it would normally be found in the VS installation. Moreover, part of my build calls `midlrt.exe` from the Windows SDK, which fails if it cannot find `cl.exe` as a C preprocessor.

Is there a way to control the use of a VS dev cmd environment for the cmake invocation, independently from the use of the CL compiler?

**Comment by gcampbell-msft** [comment, marked as answer]
Hi @tristanlabelle! 

Currently, there is not a way for you to do this explicitly. This is very closely related to an issue that has gotten a lot of attention from the community: #1836. 

This experience is something that we're prioritizing for an upcoming release. This is a known area of frustration and we are thinking and working on how to improve this for users. 

Some possible workarounds include the following: 

1. You could modify your CMakePresets.json to modify the PATH to include the things you need like ninja, and cl.exe. 
2. You could create a CMakeUserPresets.json that would help add the things you need from the dev cmd. 
3. You could launch VS Code from a dev cmd prompt.

**Reply by tristanlabelle** [reply]
Thanks for the info and possible workarounds. I also tried setting CMAKE_ASM_COMPILER to `cl` in the hope that it would trick the extension into setting up a VS dev environment, but that didn't work.

Reviewing the linked issue and issues on the repo in general, I do not see one that corresponds to this exactly. Should I file a new one?

**Reply by gcampbell-msft** [reply]
Yes, I think that would be helpful. The most helpful thing would be to create a specific issue for this, and link it to #1836, since we view this as something that would be fixed through an improvement for #1836 . 

Thanks!

**Reply by tristanlabelle** [reply]
Filed here: https://github.com/microsoft/vscode-cmake-tools/issues/3497 . Thanks!
